### ISSUE TITLE
Native joystick support and ideas

### ISSUE THREAD

**Issue by DieppedalleDavid** [Scirra/Construct-feature-requests#268]
### Reviewed guidelines

- [X] I have read and understand the suggestion guidelines

### Checked for duplicate suggestions

- [X] I checked for existing similar suggestions

### Summary

- it would be nice to add full joystick support.

so you don't have to code like you do for the keyboard, that would be a big plus.

- Be able to return the value of the joystick angle (the angle at which the left or right joystick is currently positioned). (This could be used, for example, to move a character at the same angle as the joystick, another idea would be to be able to move the character at a speed proportional to the joystick's inclination, the more it's inclined the faster the character moves).

- I don't know if it's possible to do that, but also for the left and right triggers, to know the pressure value, a bit like for joysticks.

### Possible workarounds or alternatives

yes, this can be done natively in construct with events, but it's not easy to do!

### Proposed solution

the joystick should be supported natively, like the keyboard, which can be disabled by default if you wish.

it should also be possible to configure speed and automatic rotation as for the keyboard, but with the additional option of being able to define the rotation angle in steps of 8 directions, or if there are 4 directions or 2 directions, for example:

- for 8 moves, I set automatic rotation, then choose between two other options, turning the character proportionally to the joystick (free) or in 45-degree steps (0, 45, 90, 135, 180, 225, 270, 315)

- for 4 moves, I set automatic rotation, then choose between two other options, turn the character proportionally to the joystick (free) or in 90-degree steps (0, 90, 180, 270).

- for 2 moves (left and right), I set automatic rotation, then choose between two other options, turn the character proportionally to the joystick (free) or in 180-degree steps (0, 180).

- for 2 moves (up and down), I set automatic rotation, then choose between two other options, turn the character proportionally to the joystick (free) or in steps of 180 degrees (90, 270).

Basically, if you know what I mean...

Another idea, by the way:
move the character proportionally to the joystick.
The more the joystick is tilted, the faster the character moves (configurable).

As for the joystick triggers, I don't know if it's possible to return to their position (state of depression). 

### Why is this idea important?

there's probably a lot more to it than my idea, it's just an idea, but it would simplify c3 even more.
